### PR TITLE
Persist security scan events in unified activity history

### LINKED ISSUE #60: [FEATURE] Persist security scan events in unified activity history
Parent Epic: #61
Depends on: #58
Security source: #57 / #112

## Summary

Persist **explicit security scan operations** in DustFril's existing unified local activity history so users can review when a security inspection ran, what it inspected, whether the scan completed, and what risk summary it produced.

This issue must extend the existing history model introduced by #58. Do **not** create a second security-specific log file or a parallel persistence system.

The goal is operational transparency, not SIEM-style telemetry.

## Current state

`dustfril-core` already has:

- versioned local activity-history persistence;
- `ActivityKind::{Scan, Cleanup, Security}`;
- atomic replacement behavior for `history.json`;
- legacy cleanup-history migration;
- CLI/Tauri consumers of unified history.

Security scanning already produces structured findings through Core.

The missing behavior is wiring an explicit security scan into the unified activity model with safe, stable semantics.

## Required event boundary

Record **one activity record per explicit user-visible security scan**.

Example workflow:

```text
user runs security scan
    ↓
Core security scan completes or fails
    ↓
one Security activity record
```

Do not record one activity for every internal helper call.

Do not create multiple records merely because a security scan internally performs:

- lifecycle inspection;
- lockfile inspection;
- dependency-source inspection;
- manifest parsing.

The history should represent user-visible operations, not the internal call graph.

## Event model

Reuse the existing `ActivityRecord` / `ActivityKind::Security` model.

The Security record should contain enough structured information to render a useful history entry without re-running the scan.

At minimum record:

```text
Timestamp
Target path / workspace identity
Selected ecosystem if applicable
Execution success/failure
Finding count
Highest observed risk
Finding category/rule identifier
Source file/location where useful
Sanitized reason/summary
```

A suggested conceptual shape is:

```json
{
  "kind": "security",
  "result": {
    "success": true,
    "details": {
      "path": "/workspace/project",
      "findingCount": 2,
      "highestRisk": "high",
      "findings": [
        {
          "rule": "remote-script-execution",
          "risk": "high",
          "source": "package.json",
          "reason": "Remote content is executed by a shell"
        }
      ]
    }
  }
}
```

This example is descriptive, not a requirement to rename the existing public serialized fields.

Preserve the existing history compatibility/version contract unless a version change is genuinely required.

## Sensitive-data requirements

Security logs can accidentally become a second secret store.

Do not persist raw secrets or credentials merely because they appeared in a command or workflow.

Requirements:

- do not persist environment-variable values;
- do not persist token/password/credential literals when they can be identified;
- prefer stable rule IDs, source locations, and sanitized reasons over raw command duplication;
- if an existing finding includes command text, apply a deliberate sanitization/redaction policy before persistence;
- never upload history externally.

Add tests for representative redaction behavior if raw finding text can contain sensitive values.

## Failure semantics

History persistence is **auxiliary** to the security scan.

A successful security inspection must not become unusable merely because local history cannot be written.

Examples:

- read-only app-data directory;
- corrupted/unsupported history file;
- atomic replacement failure.

The caller must receive/report the history failure, but it must not falsely change a completed security scan into an unperformed scan unless the existing product contract explicitly treats all history failures as fatal.

Follow the same non-blocking principle already used for scan/cleanup activity recording.

## Scan-result semantics

Distinguish clearly between:

```text
scanner execution failed
```

and:

```text
scanner succeeded and found security warnings
```

A finding is not automatically an operational failure.

Persist both:

- scan execution status;
- finding/risk summary.

Do not encode `success = false` merely because a High/Critical finding exists unless that behavior is explicitly documented across the Core API and CLI.

## Core / CLI / Tauri boundaries

Reusable event creation/persistence belongs in `dustfril-core`.

CLI/Tauri may:

- invoke the scan;
- invoke/trigger activity recording through the public Core API;
- present persistence errors;
- render activity records.

They must not construct incompatible security-log schemas independently.

If the desktop history view already supports generic activity records, extend it minimally to render Security entries. Do not redesign the entire desktop UI in this issue.

## Tests

Add behavior-focused regression tests covering at minimum:

### Persistence

- successful security scan records exactly one Security activity;
- record survives reload/restart;
- existing Scan/Cleanup records remain readable;
- legacy migration remains valid;
- appending a Security record does not drop previous history.

### Operation boundary

A security workflow composed of multiple internal checks still produces one activity record.

### Results

- zero findings;
- multiple findings;
- highest-risk calculation;
- failed scan operation;
- partial/unsupported checks if the current scanner reports them.

### Failure handling

- history write failure is surfaced;
- history write failure does not duplicate events on retry unexpectedly;
- security scan result remains available when auxiliary recording fails according to the documented contract.

### Sensitive data

- representative credential/token-like content is not persisted verbatim when the finding path can contain it.

Use temporary directories for persistence tests.

## Quality gates

Run:

```bash
cargo fmt --all -- --check
cargo clippy --workspace --all-targets -- -D warnings
cargo test --workspace
```

If Tauri/frontend rendering changes, run the maintained desktop frontend build.

Also run focused coverage for the changed history/security paths when practical.

## Non-goals

Do not add:

- a separate `security-history.json` store;
- cloud log upload;
- external SIEM integration;
- notifications/webhooks;
- automatic remediation;
- new scanner rules;
- online advisory lookup;
- retention/rotation infrastructure unless an existing history-size regression requires a focused fix;
- broad History-page redesign.

## Acceptance criteria

- [ ] Exactly one Security activity is recorded per explicit security scan.
- [ ] Internal sub-checks do not generate duplicate user-visible events.
- [ ] Security events reuse the versioned unified history store.
- [ ] Scan execution status and security-finding severity are represented separately.
- [ ] Sensitive values are not persisted verbatim through security findings.
- [ ] History-write failures are surfaced without silently losing the completed scan result.
- [ ] Existing Scan/Cleanup history and migration behavior remain compatible.
- [ ] Desktop/CLI consumers use the Core-owned activity representation.
- [ ] Regression tests cover persistence, duplicate prevention, failures, and sanitization.

### PR DESCRIPTION
**Pull request by mors119**
## Summary
- persist one Core-owned Security activity per explicit CLI/Tauri security scan
- record finding count, highest risk, ecosystem scope, stable rule IDs, source paths, and sanitised reasons
- record failed scans without treating findings as operational failures
- keep history writes auxiliary and surface write errors without losing completed results
- expose the Tauri security_scan command and render Security entries in the history view

## Tests
- cargo fmt --all -- --check
- cargo clippy --workspace --all-targets -- -D warnings
- cargo test --workspace --no-fail-fast
- npm run build (apps/dustfril-tauri)

Closes #60

### CODE REVIEW

File: crates/dustfril-core/src/models/history.rs [hunk 1]
```diff
@@ -79,6 +79,69 @@ impl ActivityResult {
...
+                    "reason": sanitize_text(&finding.reason),
+                })
+            })
+            .collect::<Vec<_>>();
+
+        Self::new(
+            true,
+            json!({
+                "path": sanitize_path(target_path),
+                "ecosystems": ecosystem_labels(ecosystems),
```
[the remark this answers is not in the document]
Comment (mors119) [reply crates/dustfril-core/src/models/history.rs]: Accepted with adjustment. The scope issue is valid: the previous payload persisted the raw selection, so an empty CLI selection appeared empty and Java could be misreported. I added a Core-shared `effective_security_ecosystems` helper used by both the security scanner and activity-record builder: empty means Node+Rust, explicit selections retain only supported Node/Rust entries, and duplicates are removed. The History UI now displays `None` for an explicitly unsupported-only scope. Added regression coverage. Implemented in commit `89b6485`.
